sdk/core/azure-core/src/main/java/com/azure/core/models/PollResult.java

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.models;
+
+import com.azure.core.annotation.Immutable;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/** Provides status details for long running operations. */
+@Immutable
+public final class PollResult {
+    @JsonProperty(value = "id", required = true)
+    private final String operationId;
+
+    @JsonProperty(value = "error")
+    private ResponseError error;
```

Comment:
Do we want to make "error" a template variable, so that it maybe substituted?
(as TypeSpec is a template https://github.com/Azure/typespec-azure/blob/0c1ba6f171a42571ab2d66aac2425cceec92ce6d/packages/typespec-azure-core/lib/foundations.tsp#L38)